pkg/discovery/util/pod_util.go

```diff
@@ -163,3 +165,69 @@ func podIsReady(pod *api.Pod) bool {
 	glog.Errorf("Unable to get status for pod %s", pod.Name)
 	return false
 }
+
+// GetPodInPhase finds the pod with the specific name in the specific phase (e.g., Running).
+// If pod not found, nil pod pointer will be returned without error.
+func GetPodInPhase(podClient v1.PodInterface, name string, phase api.PodPhase) (*api.Pod, error) {
+	pod, err := podClient.Get(name, meta_v1.GetOptions{})
+	if err != nil {
+		return pod, err
+	}
+
+	if pod == nil {
+		glog.V(3).Infof("Pod %s is not found", name)
+		return nil, nil
```

Comment:
`return nil, fmt.Errorf("Pod %s is not found", name)` will make it easier for invoker.

Reply:
Here, it is intended to return nil pod without error. In the caller, we want to distinguish if it's an error from api call or it's simply no such pod exist (so we can try cached name). If it is an error with api call, we don't need to try cached value while if it's pod not found, we should try other possible cases (e.g., cached name, uid).

Reply:
Should we try to find it in the cache in either case if the pod is not available via k8s API?

Reply:
That's similar to what we do now (if I understand it correctly). The caller will find the pod with the cached name first and if not found, try the original pod name. So, we need return nil error when this method doesn't find any pod.

Reply:
Anyway, I think it is not safe to use `pod != nil` as a flag. Because in case of API call fail, the result of `pod` is not defined.  Right now, it is not `nil`, but it may change.  Would rather add another flag to distinguish the different cases if necessary.